authentication.py

```diff
@@ -0,0 +1,84 @@
+import urllib
+import json
+import endpoints
+import os
+import logging
+
+from google.appengine.api import memcache
+from google.appengine.api import urlfetch
+
+_ACCEPTED_AUDIENCE_PREFIXES = os.environ['BOND_ACCEPTED_AUDIENCE_PREFIXES'].split()
+_ACCEPTED_EMAIL_SUFFIXES = os.environ['BOND_ACCEPTED_EMAIL_SUFFIXES'].split()
+_TOKENINFO_URL = 'https://www.googleapis.com/oauth2/v1/tokeninfo'
+
+
+class UserInfo:
+    def __init__(self, id, email, token, expires_in):
+        self.id = id
+        self.email = email
+        self.token = token
+        self.expires_in = expires_in
+
+    def __str__(self):
+        return 'id: {}, email: {}, token: {}, expires_id: {}'.format(self.id, self.email, self.token, self.expires_in)
+
+
+def _token_info(token):
+    result = urlfetch.fetch(
+        '%s?%s' % (_TOKENINFO_URL, urllib.urlencode({'access_token': token})))
+    if result.status_code != 200:
+        raise endpoints.InternalServerErrorException(message='Token info endpoint returned status {}: {}'.format(result.status_code, result.content))
+    return result
+
+
+def require_user_info(request_state, token_info_fn=_token_info):
+    """Get the user's info from cache or from google if not in cache, throwing unauthorized errors as appropriate
+
+    :param request_state: self.request_state from a class extending protorpc.remote.Service
+    :param token_info_fn: function to get token info, defaults to google http request, override for testing
```

Comment:
In reference to the override statement, maybe add something here to reference `_fetch_user_info` for the contract details that the `token_info_fn` must abide by.